source/lib-intro.tex

```diff
@@ -1980,189 +2164,388 @@
+\end{itemize}
+\tcode{p} has not been invalidated by
+an intervening call to \tcode{deallocate}.
+
+\pnum
+\throws
+Nothing.
+\end{itemdescr}
+
+\begin{itemdecl}
+a.max_size()
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\result
+\tcode{X::size_type}
+
+\pnum
+\returns
+The largest value that can meaningfully be passed to \tcode{X::allocate()}.
+
+\pnum
+\remarks
+Default:
+\tcode{numeric_limits<size_type>::max() / sizeof(value_type)}
+\end{itemdescr}
+
+\begin{itemdecl}
+a1 == a2
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\result
+\tcode{bool}
+
+\pnum
+\returns
+\tcode{true} only if storage allocated from each can
+be deallocated via the other.
+
+\pnum
+\throws
+Nothing.
+
+\pnum
+\remarks
+\tcode{operator==} shall be reflexive, symmetric,
+and transitive.
+\end{itemdescr}
+
+\begin{itemdecl}
+a1 != a2
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\result
+\tcode{bool}
+
+\pnum
+\returns
+\tcode{!(a1 == a2)}
+\end{itemdescr}
+
+\begin{itemdecl}
+a == b
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\result
+\tcode{bool}
+
+\pnum
+\returns
+\tcode{a == Y::rebind<T>::other(b)}
+\end{itemdescr}
+
+\begin{itemdecl}
+a != b
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\result
+\tcode{bool}
+
+\pnum
+\returns
+\tcode{!(a == b)}
+\end{itemdescr}
+
+\begin{itemdecl}
+X u(a);
+X u = a;
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\ensures
+\tcode{u == a}
+
+\pnum
+\throws
+Nothing.
+\end{itemdescr}
+
+\begin{itemdecl}
+X u(b);
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\ensures
+\tcode{Y(u) == b} and \tcode{u == X(b)}
+
+\pnum
+\throws
+Nothing.
+\end{itemdescr}
+
+\begin{itemdecl}
+X u(std::move(a));
+X u = std::move(a);
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\ensures
+The value of \tcode{a} is unchanged and is equal to \tcode{u}.
+
+\pnum
+\throws
+Nothing.
+\end{itemdescr}
+
+\begin{itemdecl}
+X u(std::move(b));
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\ensures
+\tcode{u} is equal to the prior value of \tcode{X(b)}.
+
+\pnum
+\throws
+Nothing.
+\end{itemdescr}
+
+\begin{itemdecl}
+a.construct(c, args)
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\result
+(not used)
+
+\pnum
+\effects
+Constructs an object of type \tcode{C} at \tcode{c}.
+
+\pnum
+\remarks
+Default:
+\tcode{construct_at(c,~std::forward<Args>(args)...)}
+\end{itemdescr}
+
+\begin{itemdecl}
+a.destroy(c)
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\result
+(not used)
+
+\pnum
+\effects
+Destroys the object at \tcode{c}.
+
+\pnum
+\remarks
+Default: \tcode{destroy_at(c)}
+\end{itemdescr}
+
+\begin{itemdecl}
+a.select_on_container_copy_construction()
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\result
+\tcode{X}
+
+\pnum
+\returns
+Typically returns either \tcode{a} or \tcode{X()}.
+
+\pnum
+\remarks
+Default: \tcode{return a;}
+\end{itemdescr}
+
+\begin{itemdecl}
+X::propagate_on_container_copy_assignment
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\result
+Identical to or derived from \tcode{true_type} or \tcode{false_type}.
```

Comment:
Preexisting: These probably needs to be more than just derived. Like "a type with a base characteristic ([meta.rqmts]) of `bool_constant<`_`B`_`>`".
In that case, the plain use of `true_type` in the descriptions should be changed accordingly.

Reply:
This is now #5348.